_posts/2020-10-21-zarr-v3-first-draft.md

```diff
@@ -0,0 +1,109 @@
+---
+layout: post
+title:  "Zarr V3 Spect First Draft Publication"
+date:   2020-10-21
+categories: zarr specs
+---
+
+Zarr is a format for the storage of chunked, compressed, N-dimensional arrays,
+started in 2018, initially designed for a local file system it has grown through
+the years to work on many storage infrastructure, in particular cloud storage
+like S3, google cloud storage and many others. The current specification of Zarr
+is robust and in production in many existing libraries, but there are a [number
+of reasons the pushed us to designed a new version of the
+specification](https://zarr-developers.github.io/zarr/specs/2019/06/19/zarr-v3-update.html),
+among which a better design for cloud storage, extensibility and being language
+agnostic.
+
+After many month of work we are publishing the draft of the [Zarr V3
+spec](https://zarr-specs.readthedocs.io/en/core-protocol-v3.0-dev/protocol/core/v3.0.html),
+and ask feedback on a few specific areas of interest for the specification. 
+
+We encourage you to read about the high level changes to the specifications as
+to whether this would affect your workflow in a positive or negative manner and
+send us your feedback by openeing or commenting on a issues on teh [Zarr Spec
+Repository](https://github.com/zarr-developers/zarr-specs). 
+
+You can also follow the current draft implementation, but be careful those might
+be incomplete and the spec is not final, so use with care. 
+  - CZarr/Xtensor LINK (implementation of zarr in C/C++)
+  - Zarrita
+  - Pull-Request to make the main zarr-python compatible with v3. 
+
+# Main changes to the spec with respect to v2
+
+Unlike Zarr spec v2, the spec v3 has mainly the following differences:
+
+  - Zarr V3 is a flat key-value store instead of a hierarchical store. Hierarchy
+    is implied. This make it easier to do concurrent creation of arrays in deep
+    hierarchy while avoiding the necessary and often costly step of creating
+    intermediary groups. 
+
+  - Zarr V3 has an explicit root, while v2 roots and groups could not be
+    distinguished. This mean that you can only open a zarr v3 at a specific
```

Comment:
This was one of the first things that readers commented on to me. I wonder if the implementation details could discuss walking up the implicit hierarchy to detect the root _a la_ git?

Reply:
I've clarified; having a single root is a spec difference, automatically finding the root is an implementation and user convenience. The core function can perfectly refuse to open a non-root, but the user exposed one can do all the magic they want. 

In the same way that right now a store can be just a string that start with `s3://` and will converted to an actual `Store()`